### ISSUE TITLE
No data being returned when searching for game

### ISSUE THREAD

**Issue by Jhanlon95** [Lacro59/playnite-howlongtobeat-plugin#161]
When using right click and `view howlongtobeat data` nothing is being returned from the search results. 

![image](https://user-images.githubusercontent.com/11045596/190158644-e5450019-44a3-4c7a-ae2e-986c81da549b.png)

When checking the logs the error `Web error with status code NotFound` is being returned, I debugged the code and it's coming back as a 404 page. Checking the HowLongToBeat website they appear to be searching differently, they no longer use `howlongtobeat.com/search_results.php`, instead it looks like they are now using `howlongtobeat.com/api/search` to perform searches, I'm unsure if that's what is actually causing the problems

Logs attached below
[extensions.log](https://github.com/Lacro59/playnite-howlongtobeat-plugin/files/9566457/extensions.log)
[playnite.log](https://github.com/Lacro59/playnite-howlongtobeat-plugin/files/9566458/playnite.log)

**Comment by SquishyMochi514** [comment]
I manage to make it work from the new Json API instead of digging in HTML

![image](https://user-images.githubusercontent.com/42750224/191689087-ef7d6d31-324c-40a5-bec3-36ffa57796ac.png)

I just entered fake data for VS/solo/coop. Does anyone know how that information was computed by the old php page ?
I got some data in JSON and I may have to do an additional request after the gamesearch to get more info to fit that in
Basically, isn't Solo kinda similar to main completion ? It would be helpful to get me some game ids which have a little bit of all those multiplayer information/solo information so that I can try and validate

**Comment by Jhanlon95** [comment]
Very cool! I've been trying to fix the search as too, would it be possible to see what you've done to fix it? Would be nice to know if I'm just overcomplicating things with what I'm trying to do

**Comment by SquishyMochi514** [comment]
It did in fact simplify a lot of the code using JSON instead of scrapping the rendered HTML but I think you got no choice since I'm not sure if there used to be JSON. But again we are not using a public official API even with the new /api/ so we are not backed against future random changes. There are still multiple things to validate, like all the options to push user data from Playnite to hltb. Pretty sure those are broken as well. As I didn't really know that it existed, I never used it and I will look into the code to understand and adapt all those features. Did they change their private API often in the past ? I was thinking about how we could cover that with unit tests, but we would need a test account of some sort.

**Comment by Jhanlon95** [comment]
I don't think they change that often, the API has been the first major change I've seen coming from HowLongToBeat, and even then I think that's because it's been a highly requested feature so that other websites can implement it.

I have a fork of this repo which we could both work on instead of us both working on separate repositories if that makes sense to you?

**Comment by SquishyMochi514** [comment]
I will do the fork push before the weekend. The code is still very early as I'm figuring out how to reattach and map all the different part of it.

**Comment by SquishyMochi514** [comment]
https://github.com/SquishyMochi514/playnite-howlongtobeat-plugin

**Comment by Mimo433** [comment]
@SquishyMochi514 Hey there! any chance you could upload a pext for us with the applied fixes or make an easy tutorial on how to fix it? I'm not a programmer myself and I don't understand code, sorry. But I'm sure me and many others who are struggling rn would appreciate simplified way of fixing it, thank you.

**Comment by SquishyMochi514** [comment]
Fixes are being applied on my fork (if you ever want to take a look at it). Finishing the pushing of HLTB data from the plugin right now. If you can hold on a little bit, that will be done pretty soon.

**Comment by Mimo433** [comment]
@SquishyMochi514 I did look at it but I was completely lost. I didn't know how to use the fork to apply the fixes at all. I admire how you guys can understand these stuff. If there's away to use the fork to apply them enlighten me please! If not, I'll happily thank you for your efforts and wait for the release.

**Comment by dobrowski** [comment]
@Mimo433 and @SquishyMochi514 It looks like @Lacro59 is working on the master branch to address the API changes with more recent commits and merges.  You may want to confer on that to make sure the work isn't duplicated/redundant.  Thanks for working to get this plug-in in tiptop shape.   I really appreciate it.

**Comment by Lacro59** [comment]
fixed with last release

**Comment by Braxusz** [comment]
I would like to report that this looks to be happening again today.
